web/third/marked/lib/marked.d.ts

```diff
@@ -0,0 +1,42 @@
+/*
+This is an incomplete stub, which we include here because
+we only use parts of the marked API and are using a fork
+of the upstream library.
+*/
+
+import type {ParseOptions} from "../../../src/markdown";
+
+export class Renderer {
+    code: (code: string) => string;
+    link: (href: string, title: string, text: string) => void;
+    br: () => string;
+}
+
+export type RegExpOrStub = RegExp | {
+    exec: () => boolean;
+};
+
+type MarkedOptions = ParseOptions & {
+    userMentionHandler: (mention: string, silently: boolean) => string | undefined;
+    groupMentionHandler: (name: string, silently: boolean) => string | undefined;
+    silencedMentionHandler: (quote: string) => string;
+};
+
+type MarkedStub = {
+    (src: string, opt: MarkedOptions): string;
+    Lexer: {
+        rules: {
+            tables: Record<string, RegExpOrStub>;
```

Comment:
We should be more specific about which keys there are—this isn’t used as a map with arbitrary string keys.